docs/using-kontena/users.md

```diff
@@ -4,23 +4,41 @@ title: Users
 
 # Users
 
-Kontena has built-in role-based user management. The first user that logs in to Kontena Master will be assigned to the  `master_admin` role. The Master admin can invite new users to the Master server and assign users to the `master_admin` or `grid_admin` roles. The Master admin and Grid admin can add and remove users from a Grid.
+Kontena uses roles to grant access to specific grid operations to master users.
+New Kontena masters are initially provisioned with a local `admin` user using the `INITIAL_ADMIN_CODE`, which can be used to invite other users and grant them roles.
+Users with the `master_admin` role are equivalent to the initial `admin` user, and other more specific roles can be used grant more limited access.
 
+The Kontena master API uses OAuth2 tokens for [Authentication](`authentication`), and users are identified by email address. The default OAuth2 provider for plugin-provisioned masters is the [Kontena Cloud](authentication#configuring-kontena-cloud-as-the-authentication-provider).
+
+Each user belongs to a specific list of grids, and users can only list and access grids that they belong to.
+The local `admin` user and `master_admin` role can access any grids.
+Kontena roles are assigned to users, and apply to any grids that the user is a member of.
+By default, users only have access to the secrets, volumes, stacks, services etc. within a grid.
 
 ## Roles
 
-**master_admin**
-  * can invite users to Kontena Master
-  * can manage user roles
-  * can manage all Grids and their users
+### `master_admin`
+  * All `users_admin` permissions
+  * Add and remove user roles
+  * Manage grids (create, remove)
+  * All `grid_admin` permissions for all grids
+
+### `users_admin`
+  * Invite and remove users
+  * Manage grid users (only user grids)
+
+### `grid_admin`
+
+  * Manage grid (update)
+  * Manage grid users
+  * Manage grid host nodes (create, reset tokens)
```

Comment:
BTW: `kontena node rm` should probably also require `grid admin`? What about the `kontena node update` labels, and  #2306 availability?